torch/_subclasses/meta_utils.py

```diff
@@ -371,19 +407,48 @@ def is_c_of_r(complex_dtype, real_dtype):
                         # recreate this situation.
                         def _view_from_base(base, t):
                             if t.is_nested:
-                                # Nested tensors do not support as_strided, and
-                                # hence,always have _view_func available.
-                                #
-                                # The unsafe version of _view_func omits
-                                # checking whether the base passed in has the same
-                                # metadata as the original base the view_func
-                                # was originally executed with. (1) It is OK here,
-                                # because we're calling it on the meta-ified base,
-                                # so the metadata is guaranteed to be the same.
-                                # (2) It is necessary because we don't actually
-                                # want to guard on the base's metadata here.
-                                return t._view_func_unsafe(base)
+                                if base.is_nested:
+                                    # Nested tensors do not support as_strided, and
+                                    # hence,always have _view_func available.
+                                    #
+                                    # The unsafe version of _view_func omits
+                                    # checking whether the base passed in has the same
+                                    # metadata as the original base the view_func
+                                    # was originally executed with. (1) It is OK here,
+                                    # because we're calling it on the meta-ified base,
+                                    # so the metadata is guaranteed to be the same.
+                                    # (2) It is necessary because we don't actually
+                                    # want to guard on the base's metadata here.
+                                    return t._view_func_unsafe(base)
+                                else:
+                                    from torch._dynamo.source import AttrSource
+
+                                    # base is ostensibly a standard dense tensor
+                                    # fake-ify offsets then return view
+                                    offsets_src = AttrSource(source, "_offsets")  # type: ignore[arg-type]
+                                    fake_offsets = callback(
+                                        lambda: empty_create(t._offsets, offsets_src)
+                                    )
```

Comment:
Do we also want to consider the case where base is a view? In that case would it be better to call `self.meta_tensor` in order to make sure that view is recursively metaified as well?

Reply:
can base ever be a view? my understanding is that base is the basest of bases so never a view
e.g. (base -> view -> view of view); view of view's _base is the OG base